clippy_lints/src/write.rs

```diff
@@ -250,10 +267,39 @@ impl EarlyLintPass for Write {
     }
 }
 
+/// Given a format string that ends in a newline and its span, calculates the span of the newline.
+fn newline_span(fmt_str: &FmtStr) -> Span {
```

Comment:
This function can be implemented as a method of `FmtStr` now:
```rust
impl FmtStr {
    fn newline_span(&self) -> Span { ... }
}
```

Reply:
Done.